### ISSUE TITLE
Implement EuclideanDistance between Line and Polygon

### ISSUE THREAD

**Issue by JivanRoquet** [georust/rust-geo#224]
Unless I'm mistaken, this is not currently possible to compute an euclidean distance between the following combinations:

- Line <> Polygon
- Polygon <> Polygon
- Line <> Line

Implementing the `EuclideanDistance` trait for the following cases would be a nice addition to the library capabilities:

```rust
EuclideanDistance<T, Polygon<T>> for Line<T>
EuclideanDistance<T, Polygon<T>> for Polygon<T>
EuclideanDistance<T, Line<T>> for Line<T>
```

Starting with `Line, Line` combination would be a good idea, as the other two combinations could derive from that, possibly adding optimisations.

**Comment by urschrei** [comment]
All of these (IIRC) were [merged yesterday](https://github.com/georust/rust-geo/pull/219), along with `LineString`-`LineString`, and `LineString`-`Polygon`. Note the "fast path" `Polygon`-`Polygon` euclidean distance optimisation. I imagine we'll cut a new release this week!

**Comment by frewsxcv** [comment]
I'll cut a release after this next PR merges

**Comment by frewsxcv** [comment]
just published version 0.9.0 of the geo crate 🍻
